source/plugin/data/keys.rst

```diff
@@ -0,0 +1,109 @@
+==========
+Using Keys
+==========
+
+Getting and offering data using a key
+=====================================
+
+Since accessing even a single point of data going from ``DataHolder`` to the correct ``DataManipulator`` tended to
+be quite tedious and bloat both the code and the import list, a more direct way of accessing values via ``Key``\ s
+was devised. Let's just start out with an example.
+
+**Code Example: Healing a data holder, if possible**
+
+.. code-block:: java
+
+    import org.spongepowered.api.data.DataHolder;
+    import org.spongepowered.api.data.key.Keys;
+
+    public void heal(DataHolder target) {
+        if (target.supports(Keys.HEALTH)) {
+            double maxHealth = target.getOrNull(Keys.MAX_HEALTH);
+            target.offer(Keys.HEALTH, maxHealth);
+        }
+    }
+
+Now for the details of the above function.
+
+The first line checks if our given data holder possesses health. Only if he does, he can be healed after all.
+Since a data holder can not have health without having a maximum health and vice versa, a check for one of the
+keys using the ``supports()`` method suffices.
+
+The second line uses the ``getOrNull()`` function to ask the data holder for its maximum health. Besides
+``getOrNull()``, the methods ``get()`` and ``getOrElse()`` exist, all of which accept a ``Key`` as their first
+parameter. Generally, ``get()`` should be used. It will return an ``Optional`` of the data requested or
```

Comment:
"Generally, ``get()`` should be used. It..." --> "Generally, ``get()`` should be used, which will..."